### PR TITLE
Fixes and changes

### PR DESCRIPTION
**Pull request by Njs41**
- Added horses
- Fix issue with setting entities as babies.
- Fixed chicken's voiceboxes so they make chicken sounds instead of cat sounds.
- Added methods for dealing with nbt tags to help with fixing admintoolbox's bindPermanant command.

### CODE REVIEW

File: src/no/runsafe/framework/api/entity/animals/horses/IAbstractHorse.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package no.runsafe.framework.api.entity.animals.horses;
+
+import no.runsafe.framework.api.entity.IAnimal;
+import no.runsafe.framework.api.entity.ITameable;
+
+public interface IAbstractHorse extends IAnimal, ITameable
```
Comment (mortenn) [inline src/no/runsafe/framework/api/entity/animals/horses/IAbstractHorse.java]: I'd say drop the Abstract part of the name here.. :p

Comment (Njs41) [inline src/no/runsafe/framework/api/entity/animals/horses/IAbstractHorse.java]: In 1.11 and up horse types/variants are different classes and Horses extend AbstractHorse, I set it up that way to make it easier to update horses to 1.11 in the future.

Comment (mortenn) [inline src/no/runsafe/framework/api/entity/animals/horses/IAbstractHorse.java]: I see an IHorse and an IChestedHorse, but I don't see any usage of the chested one, and Abstract makes for a poor name. I see no reason to reflect poorly named classes upstream when we can do better.

Comment (mortenn) [inline src/no/runsafe/framework/api/entity/animals/horses/IAbstractHorse.java]: Maybe call them IHorse, IRideableHorse and IChestedHorse? .. though I'm not super happy about the gramatics of IChestedHorse, I can live with that, I suppose

File: src/no/runsafe/framework/minecraft/entity/animals/RunsafeSheep.java [hunk 2]
```diff
@@ -48,5 +56,6 @@ public Sound getIdleSound()
 		return Sound.Creature.Sheep.Idle;
 	}
 
+	private Random random = new Random();
```
Comment (mortenn) [inline src/no/runsafe/framework/minecraft/entity/animals/RunsafeSheep.java]: Something feels wrong about keeping a rng on every sheep instance just to facilitate `setRandomColour()` - make it a static?

File: src/no/runsafe/framework/minecraft/entity/animals/horses/RunsafeNormalHorse.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		normalHorse.setStyle(values[this.random.nextInt(values.length)]);
+	}
+
+	@Override
+	public Horse getRaw()
+	{
+		return normalHorse;
+	}
+
+	private Random random = new Random();
```
Comment (mortenn) [inline src/no/runsafe/framework/minecraft/entity/animals/horses/RunsafeNormalHorse.java]: Same thing here

**Pull request merged**